Blame option description was changed and example was added. (#5536)
MicrosoftDocs/visualstudio-docs: docs/test/configure-unit-tests-by-using-a-dot-runsettings-file.md

To customize any other type of diagnostic data adapters, use a [test settings file](../test/collect-diagnostic-information-using-test-settings.md).

### Blame data collector

```xml
<DataCollector friendlyName="blame" enabled="True">
  <Configuration>
    <ResultsDirectory>C:\TestResults</ResultsDirectory>
    <CollectDump CollectAlways="true" DumpType="mini" />
  </Configuration>
</DataCollector>
```

> This option is helpful in isolating the problematic test causing test host crash. It creates an output file (`Sequence.xml`) in directory specifying by ***ResultsDirectory*** node, that captures the order of execution of test before the crash. 

Reviewer: ```suggestion
This option is helpful in isolating the problematic test causing test host crash. It creates an output file (*Sequence.xml*) in the directory specified by the ResultsDirectory node, which captures the order of execution of the test before the crash. 
```

> This option is helpful in isolating the problematic test causing test host crash. It creates an output file (*Sequence.xml*) in the directory specified by the ResultsDirectory node, which captures the order of execution of the test before the crash. 

Reviewer 2: This setting would imply using blame collector to take a hang dump.

These are the settings that are equivalent to --blame:

```xml
<RunSettings>
    <LoggerRunSettings>
        <Loggers>
            <Logger friendlyName="blame" enabled="True" />
        </Loggers>
    </LoggerRunSettings>
    <DataCollectionRunSettings>
        <DataCollectors>
            <DataCollector friendlyName="blame" enabled="True">
            </DataCollector>
        </DataCollectors>
    </DataCollectionRunSettings>
</RunSettings>
```

The sequence file will end up in TestResults that folder that is automatically generated. The one mentioned in the settings is just a temp path.

Author: Is the role of blame option to be able to detect crashed tests and put them in Sequence.xml file? So what is difference with your example? About ResultsDirectory, we can specify folder where we want to get results, by default, as you said, it would create TestResults folder in working directory, so why do you call it 'temp path'?

Reviewer 3: @nohwnd's feedback makes sense to me. Since the Sequence.xml's default location is the TestResults folder, we don't need to specify it in runsettings. This helps keep the example as simple as possible. (It is specified in other runsettings examples to show how to set a path, but we shouldn't include it in all the examples when it's not needed.)

I don't think we want the "golden path" use case to always collect a crash dump. I'd predict most users just want to know which test crashed the process.

Author: Got it. Example simplified.

Reviewer 2: The TestResults path in the Blame collector settings is place where the dump file gets placed temporarily, and then it is moved to the real TestResults folder. It is confusing. New version of TP uses the local  temp folder.

You are right, the role of blame option **is** detecting the tests that crashed the test host. But the options you are showing are doing more, they enable taking a memory dump when the testhost process crashed.

> When you choose to collect dump, by default, a mini dump will be collected on a crash. You may also choose to override this default behaviour by some optional parameters: 

Reviewer: ```suggestion
When you choose to collect a dump, by default, a mini dump will be collected on a crash. You may also choose to override this default behavior by some optional parameters: 
```

|**InProcMode**|false|If you want your tests to be run in the same process as the MSTest adapter, set this value to **true**. This setting provides a minor performance gain. But if a test exits with an exception, the remaining tests don't run.|
|**AssemblyResolution**|false|You can specify paths to additional assemblies when finding and running unit tests. For example, use these paths for dependency assemblies that aren't in the same directory as the test assembly. To specify a path, use a **Directory Path** element. Paths can include environment variables.<br /><br />`<AssemblyResolution>  <Directory Path="D:\myfolder\bin\" includeSubDirectories="false"/> </AssemblyResolution>`|

## Specifying environment variables in *.runsettings* file

> The environment variables can be set in *.runsettings* file which can directly interact with the test host. Specifying environment variables in the *.runsettings* file is necessary to support non-trivial projects that require settings environment variables like *DOTNET_ROOT*. These variables are set while spawning the test host process, thus will be available in the host.

Reviewer: ```suggestion
The environment variables can be set in the *.runsettings* file, which can directly interact with the test host. Specifying environment variables in the *.runsettings* file is necessary to support non-trivial projects that require setting environment variables like *DOTNET_ROOT*. These variables are set while spawning the test host process, and so will be available in the host.
```

#### Example

> Below is a sample runsettings for passing environment variables :

Reviewer: ```suggestion
The following is a sample runsettings for passing environment variables :
```

```xml
<?xml version="1.0" encoding="utf-8"?>
<!-- File name extension must be .runsettings -->
<RunSettings>
  <RunConfiguration>
    <EnvironmentVariables>
      <!-- List of environment variables we want to set-->
      <DOTNET_ROOT>C:\ProgramFiles\dotnet</DOTNET_ROOT>
      <SDK_PATH>C:\Codebase\Sdk</SDK_PATH>
    </EnvironmentVariables>
  </RunConfiguration>
</RunSettings>
```

> The **RunConfiguration** node should contain a **EnvironmentVariables** node. The different environment variables can be specified as element name and it's value.

Reviewer: ```suggestion
The RunConfiguration node should contain a EnvironmentVariables node. The different environment variables can be specified as element name and it's value.
```

> [!NOTE]

>    > Since these environment variables should always be set when the test host is started, the tests should always run in a separate process. For this, the */InIsolation* flag will be set when there are environment variables so that the test host is always invoked.

Reviewer: ```suggestion
> Because these environment variables should always be set when the test host is started, the tests should always run in a separate process. For this, the */InIsolation* flag will be set when there are environment variables so that the test host is always invoked.
```
